pkg/virt-config/virt-config.go

```diff
@@ -74,8 +74,8 @@ const (
 	// Default REST configuration settings
 	DefaultVirtHandlerQPS         float32 = 5
 	DefaultVirtHandlerBurst               = 10
-	DefaultVirtControllerQPS      float32 = 20
-	DefaultVirtControllerBurst            = 30
+	DefaultVirtControllerQPS      float32 = 400
+	DefaultVirtControllerBurst            = 600
```

Comment:
These values are quite huge. I would be for increasing the default values but I still think we need to keep everything sane. If we have lots of requests coming in, we don't want to overwhelm the API server too much - note that we are not the only ones who are making the requests, and when there is a lot of churn, then there is probably lots of other components doing requests too (for example `virt-handler`s which could be hundreds of them in the cluster).

Therefore, we should throttle ourselves, at least a little bit. I think setting QPS to 50 and Burst to 70 should be more than enough. If the user has better hardware, topology and scalability, they should set their own values. But I don't think we should go higher than that for default ones.

I'm happy to hear more thoughts on that.

Reply:
I understand this concern and that's why I wrote a very large review! Note that, this is necessary to allow KubeVirt to scale accordingly to the load with a good throughput and low latency!

To analyze whether we have overloaded the virt-api, we need to look at the `apiserver_current_inflight_requests{}` metric, which shows that even increasing the virt-controller to a very high number, the number of current in-flight requests is less than 40req/s, where the maximum is 500req/s by default.

So we're a long way from the maximum of 500 of current in-flight requests, and increasing to 400/600 has a huge impact on virt controller performance. I have carefully analyzed that...

Also node that, this QPS is shared between all internal controllers in virt-controller.

If we really need a smaller number (which I don't think is the case), we could go then with the second best scenario of 200/400....

Reply:
But here you assume that the only thing that is happening on the cluster is KubeVirt handling the requests. Also not sure if looking at `apiserver_current_inflight_requests` metric is a good measure. Because it will only shoot up when things will start looking very badly on apiserver side (for example when we max out the CPU with huge load). Also you assume some hardware that there is available for apiserver/etcd and the size of the cluster.

I'm not saying that your conclusions are wrong but I feel like they are isloated. If I test everything on cluster with 1 Node I can conclude that I could increase the QPS and Burst value of `virt-handler` to let's say 100. But when you talk about cluster with 1000 Nodes then this value is becoming absurdly high for `virt-handler` and now you allow things go really bad when there is huge number of requests coming in.

So what I'm saying is that we should be really conservative for these values. Of course there is no single best number that's why it's default. Therefore, I would set it somewhere in between worst case scenario (huge number of Nodes and requests in the cluster and slow apiserver) and best performance (value which wouldn't cause performance issues in most of the cases). And I don't think 400/600 is reasonable in this sense.

Reply:
Thanks @marceloamaral for this discussion. I think it makes sense. LGTM